ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/transaction_service.proto

```diff
@@ -91,6 +91,11 @@ message GetTransactionByEventIdRequest {
   // Required
   repeated string requesting_parties = 3;
 
+  // This flag indicates whether the response should contain a TransactionTree (in case of false)
+  // or the flat Transaction (in case of true).
+  // Optional
+  bool return_flat_transaction = 4;
```

Comment:
i don't quite like this behaviour, why can't we have separate service endpoints instead?

Reply:
How would you suggest migrating to new names? Change `GetTransactionByEventId` to return flat transactions, and add a new method `GetTransactionTreeByEventId`? Unfortunately that original mixed naming (using `Transaction` when we're actually returning a `TransactionTree`) doesn't really play into our favor right now :(

Reply:
fair enough, discussed the trade-off offline

Reply:
Conclusion: we rather want to have the functionality now with sub-par naming (and sufficient documentation to mitigate that), than wait for a new version of the Ledger API to land.